src/averageAges.js

```diff
@@ -37,7 +43,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person =>
+    withChildren
+      ? people.find(child => person.name === child.mother)
+      : person.sex === 'f'
+  );
+
+  const age = women.map((woman) => woman.died - woman.born);
+
+  const avarage = age.reduce((sum, years) => sum + years, 0) / age.length;
+
+  return avarage;
```

Comment:
```suggestion
  return averageAges;
```
I think it's better to specify that it's average of ages